litebird_sim/mapmaking/common.py

```diff
@@ -418,3 +429,73 @@ def estimate_cond_number(
     max_abs_eval = max(max(eval0, eval1), eval2)
 
     return (max_abs_eval / min_abs_eval, True)
+
+
+def _build_mask_time_split(
+    time_split: str,
+    obs_list: List[Observation],
+):
+    time_mask = []
+
+    if time_split == "full":
+        for cur_obs in obs_list:
+            time_mask.append(np.ones(cur_obs.n_samples, dtype=bool))
+    elif time_split == "odd":
+        for cur_obs in obs_list:
+            mask = np.zeros(cur_obs.n_samples, dtype=bool)
+            mask[0::2] = True
+            time_mask.append(mask)
+    elif time_split == "even":
+        for cur_obs in obs_list:
+            mask = np.zeros(cur_obs.n_samples, dtype=bool)
+            mask[1::2] = True
+            time_mask.append(mask)
+    elif time_split == "year1":
+        for cur_obs in obs_list:
+            if isinstance(cur_obs.start_time_global, astropy.time.Time):
+                t_i = cur_obs.start_time_global.cxcsec
+            else:
+                t_i = cur_obs.start_time_global
+            time_mask.append((cur_obs.get_times() - t_i) < t_y1_sec)
+    elif time_split == "year2":
+        for cur_obs in obs_list:
+            if isinstance(cur_obs.start_time_global, astropy.time.Time):
+                t_i = cur_obs.start_time_global.cxcsec
+            else:
+                t_i = cur_obs.start_time_global
+            time_mask.append(
+                ((cur_obs.get_times() - t_i) >= t_y1_sec)
+                * ((cur_obs.get_times() - t_i) < t_y2_sec)
+            )
+    elif time_split == "year3":
+        for cur_obs in obs_list:
+            if isinstance(cur_obs.start_time_global, astropy.time.Time):
+                t_i = cur_obs.start_time_global.cxcsec
+            else:
+                t_i = cur_obs.start_time_global
+            time_mask.append(
+                ((cur_obs.get_times() - t_i) >= t_y2_sec)
+                * ((cur_obs.get_times() - t_i) < t_y3_sec)
+            )
+    else:
+        print("Time split not recognized!!!")
+
+    return time_mask
+
+
+def _build_mask_detector_split(
+    detector_split: str,
+    obs_list: List[Observation],
+):
+    detector_mask = []
+
+    if detector_split == "full":
+        for cur_obs in obs_list:
+            detector_mask.append(np.ones(cur_obs.n_detectors, dtype=bool))
+    elif detector_split[0:6] == "wafer_":
+        for cur_obs in obs_list:
+            detector_mask.append(cur_obs.wafer == detector_split[6:9])
```

Comment:
Small question: wouldn't be clearer to use just the name of the wafer without prepending `wafer_`? It would be probably easier for the user to remember too.

Reply:
The tag `wafer_` is there for selecting the right attribute of `cur_obs` to check. In this example if `detector_split = "wafer_L01"` that piece of code compares `cur_obs.wafer` with `detector_split[6:9]`. If instead we want to select detectors per pixel type, we can use `detector_split = "pixtype_LP1"` and compare `cur_obs.pixtype` with `detector_split[8:11]` and so on... Does it seem reasonable?

Reply:
Ah, I see, ok.